R/compile_report.R

```diff
@@ -231,15 +234,46 @@ compile_report <- function(file, quiet = FALSE, factory = getwd(),
   }
 
   dir.create(output_dir, FALSE, TRUE)
-
+  
+  output_files <- list()
   for (file in new_files) {
     destination <- file.path(output_dir, file)
     move_file(file, destination)
+    i <- match(file, new_files) #gets index
+    output_files[[i]] <- destination
   }
-
+  
   ## remove empty directories
```

Comment:
Fine, though to avoid having to match it might be more straightforward to iterate over the length of `new_files`, e.g.

```r
output_files <- vector(length(new_files), mode = "list")
for (i in seq_along(new_files)) {
  file <- new_files[i]
  destination <- file.path(output_dir, file)
  move_file(file, destination)
  output_files[[i]] <- destination
}
```

Reply:
Note that it won't change much for this code, but the pre-allocation proposed is faster for large outputs than making the output grow.